arch/arm/src/sama5/hardware/_sama5d2x_memorymap.h

```diff
@@ -517,9 +520,10 @@
 #define SAM_SFRBU_VBASE          (SAM_PERIPHC_VSECTION+SAM_SFRBU_OFFSET)
 #define SAM_CHIPID_VBASE         (SAM_PERIPHC_VSECTION+SAM_CHIPID_OFFSET)
 
-#define SAM_PIOA_VBASE           (SAM_PERIPHA_VSECTION+SAM_PIO_OFFSET)
-#define SAM_PIOB_VBASE           (SAM_PERIPHB_VSECTION+SAM_PIO_OFFSET)
-#define SAM_PIOC_VBASE           (SAM_PERIPHC_VSECTION+SAM_PIO_OFFSET)
+#define SAM_PIOA_VBASE           (SAM_PIO_VBASE)
+#define SAM_PIOB_VBASE           (SAM_PIO_VBASE+SAM_PIOB_OFFSET)
+#define SAM_PIOC_VBASE           (SAM_PIO_VBASE+SAM_PIOC_OFFSET)
+#define SAM_PIOD_VBASE           (SAM_PIO_VBASE+SAM_PIOD_OFFSET)
```

Comment:
Maybe better to add `SAM_PERIPHB_VSECTION`, `SAM_PERIPHC_VSECTION` and `SAM_PERIPHD_VSECTION` just from style perspective? And define:
```
#  define SAM_PIO_OFFSET         0x00038000 /* 0x00038000-0x0003bfff: PIOA-D */
#  define SAM_PIOA_OFFSET        0x00038000 /* PIOA */
#  define SAM_PIOB_OFFSET        0x00038040 /* PIOB */
#  define SAM_PIOC_OFFSET        0x00038080 /* PIOC */
#  define SAM_PIOD_OFFSET        0x000380c0 /* PIOD */
```
?

Reply:
I did wonder about that, and happy to take that suggestion on board. Will change it on Monday and re-push, once I've worked through all the Kconfig blunders you've found <embarrassed>

Reply:
Also please squash all `Update arch/arm/src/sama5/hardware/sam_adc.h` commits into one

Reply:
I am still nervous of git operations, especially command line operation as I use GitHub Desktop.

What is the method to do this, please?

Reply:
First of all if you are new to git and do not want to loose all your changes then I would recommend you to go with the next steps:
```
git checkout SAMA5D2-PIO-errors
git checkout -b SAMA5D2-PIO-errors-backup     # <- just checkout new branch, to save what you have now. you can push if you want of keep those changes locally
git checkout SAMA5D2-PIO-errors     # <- get back to branch for this PR
git rebase -i master       # <- to trigger interactive rebase. Here you can reorder, update, squash, etc. your commits
git rebase --continue     # <- will be needed after each step is finished until you reach the end
git push -f origin SAMA5D2-PIO-errors    # <- this will "force-push" your branch. It meats that Github version will be overwritten with your local change in SAMA5D2-PIO-errors branch, but SAMA5D2-PIO-errors-backup will still contain old code, so you can recover from it in case anything goes wrong
```

Reply:
Thank you!

Reply:
Found GitHub desktop supports this, so hopefully that's done now...

Reply:
Discovered there are #defines for PIOA/B/C/D VBASE address in _sama5d2_pio.h, so the fix will make use of those now. i.e.

#define SAM_PIOA_VASE       SAM_PIO_IOGROUPA_VBASE
etc.